### ISSUE TITLE
Unusual floating point exception in frank-kamenetskii viscosities.

### ISSUE THREAD

**Issue by alarshi** [geodynamics/aspect#3762]
I noticed a very strange behavior while working on the diffusion strain rates. The new viscosities in the visco_plastic.cc computed for frank_kamenetskii cases throws a floating-point exception because it computes the term ''temperature/reference_temperature" and by default the reference temperature = this->get_adiabatic_surface_temperature(), is zero. All the existing tests throw the same error. But, when I discussed this issue with @erinheilman  and @naliboff , the tests pass even with the default adiabatic surface temperature = 0 on her machine. I am not sure what's going on because the function to compute frank_kamenetskii is always called and the reference_temperature = 0, by default.  Therefore, we should get the floating-point exception but it passed all the tests.

**Comment by naliboff** [comment]
I'm confused about this as well. The lines in question from frank_kamenetskii.cc are:
```
        const double reference_temperature = this->get_adiabatic_surface_temperature();

        const double viscosity_frank_kamenetskii = prefactors_frank_kamenetskii[composition] * std::exp(viscosity_ratios_frank_kamenetskii[composition] * 0.5 * (1.0-temperature/reference_temperature));
```
By default, the adiabatic surface temperature is set to 0, which should have thrown multiple errors for all the default visco_plastic tests.

**Comment by tjhei** [comment]
Well, it looks like the current tester does not have floating point exceptions enabled:
```
-- Performing Test HAVE_FP_EXCEPTIONS - Failed
-- No support for feenableexcept(), disabling runtime floating point exception checks.
```
I can look into that, but that explains why the official tester is not complaining. It depends on the machine/compiler/etc. if we have support for this check.

That said, we should avoid the division by zero.

**Comment by gassmoeller** [comment]
I think there are separate problems here:
- The FK viscosity should not be computed if it is not needed (I did not notice that when reviewing #3698), this is a simple fix.
- The FK should have an `AssertThrow` that makes sure the reference temperature is not 0, since FK divides by the reference temperature there is no way around it, to use FK the adiabatic surface temperature (=reference temperature) has to be non-zero.
- Our default for adiabatic surface temperature of 0 is poorly chosen, but this can hardly be changed, because so much else depends on it.

**Comment by naliboff** [comment]
```
- The FK viscosity should not be computed if it is not needed (I did not notice that when reviewing #3698), this is a simple fix.
- The FK should have an AssertThrow that makes sure the reference temperature is not 0, since FK divides by the reference temperature there is no way around it, to use FK the adiabatic surface temperature (=reference temperature) has to be non-zero.
```
Agree with the above. I'll work on a fix.

**Comment by naliboff** [comment]
See proposed fix in #3766.

**Comment by gassmoeller** [comment]
Fixed in #3766.
